docs/data-source.md

```diff
@@ -0,0 +1,131 @@
+Retcon Data Sources
+===================
+
+This document describes the interface to be implemented by a Retcon data source
+program.
+
+Overview
+--------
+
+A *data source* is a program which Retcon can invoke when it needs to interact
+with an external system. Each invocation of a data source processes a single
+operation and they can be implemented in any language.
+
+Data sources must implement the four CRUD operations:
```

Comment:
Maybe SHOULD over MUST here? You get partial functionality if you implement some of the CRUD operations.